Résultats du premier run push button de far ood sur hmdb51

In [52]:
import numpy as np
#import matplotlib.pyplot as plt
import os

# from metrics import compute_all_metrics, acc
from sklearn import metrics
import torch
import pprint

In [2]:
path = "/data/maouche/MultiOOD/EPIC-rgb-flow/saved_files/id_HMDB_ood_EPIC_conf_a2d_npmix_best_eval.npy"
data = np.load(path)
print(data.shape)
print(data)


(4871,)
[0.44259763 0.11941737 0.37215316 ... 0.23253585 0.40844247 0.7723148 ]


**Glossaire des suffixes**

- conf : scores de confiances, cad le max du softmax
- label : les ground truth
- output : les logits tels que retournés par le modèle pour chaque donnée
- feature : feature maps, peut-être à la dernière couche


In [13]:
dir = "/data/maouche/MultiOOD/EPIC-rgb-flow/saved_files"
for file in os.listdir(dir):
    if file.endswith(".npy"):
        data = np.load(path:=os.path.join(dir, file))
        print(path)
        print(data.shape)

/data/maouche/MultiOOD/EPIC-rgb-flow/saved_files/id_HMDB_ood_EPIC_output_a2d_npmix_best_eval.npy
(4871, 43)
/data/maouche/MultiOOD/EPIC-rgb-flow/saved_files/id_HMDB_ood_EPIC_feature_a2d_npmix_best_eval.npy
(4871, 4352)
/data/maouche/MultiOOD/EPIC-rgb-flow/saved_files/id_HMDB_ood_EPIC_conf_a2d_npmix_best_eval.npy
(4871,)
/data/maouche/MultiOOD/EPIC-rgb-flow/saved_files/id_HMDB_ood_EPIC_label_a2d_npmix_best_eval.npy
(4871,)
/data/maouche/MultiOOD/EPIC-rgb-flow/saved_files/id_HMDB_ood_EPIC_pred_a2d_npmix_best_eval.npy
(4871,)


In [ ]:
dir = "/data/maouche/MultiOOD/HMDB-rgb-flow/saved_files"
names = []
sizes = []
for file in os.listdir(dir):
    if file.endswith(".npy"):
        data = np.load(path:=os.path.join(dir, file))
        names.append(path)
        sizes.append(data.shape)

name_size = list(zip(names, sizes))
sorted(name_size, key=lambda x:x[1][0], reverse=False)

[('/data/maouche/MultiOOD/HMDB-rgb-flow/saved_files/id_HMDB_pred_a2d_npmix_best_val.npy',
  (843,)),
 ('/data/maouche/MultiOOD/HMDB-rgb-flow/saved_files/id_HMDB_label_a2d_npmix_best_val.npy',
  (843,)),
 ('/data/maouche/MultiOOD/HMDB-rgb-flow/saved_files/id_HMDB_output_a2d_npmix_best_val.npy',
  (843, 43)),
 ('/data/maouche/MultiOOD/HMDB-rgb-flow/saved_files/id_HMDB_conf_a2d_npmix_best_val.npy',
  (843,)),
 ('/data/maouche/MultiOOD/HMDB-rgb-flow/saved_files/id_HMDB_feature_a2d_npmix_best_val.npy',
  (843, 4352)),
 ('/data/maouche/MultiOOD/HMDB-rgb-flow/saved_files/id_HMDB_feature_a2d_npmix_best_test.npy',
  (877, 4352)),
 ('/data/maouche/MultiOOD/HMDB-rgb-flow/saved_files/id_HMDB_pred_a2d_npmix_best_test.npy',
  (877,)),
 ('/data/maouche/MultiOOD/HMDB-rgb-flow/saved_files/id_HMDB_label_a2d_npmix_best_test.npy',
  (877,)),
 ('/data/maouche/MultiOOD/HMDB-rgb-flow/saved_files/id_HMDB_conf_a2d_npmix_best_test.npy',
  (877,)),
 ('/data/maouche/MultiOOD/HMDB-rgb-flow/saved_files/id_HMDB_outp

# Signification des suffixes

Les fichiers **pred** contiennent les classes prédites par le modèle. Il y a 843 fichiers dans le split val

In [3]:
path = "/data/maouche/MultiOOD/HMDB-rgb-flow/saved_files/id_HMDB_pred_a2d_npmix_best_val.npy"
data = np.load(path)
print(data.shape)
print(data)

(843,)
[ 0  0  0  0  0 34  0  0  0  8  0  0  0  0  0  0  1  1  1  1  1  1  1  1
  1 10 10  1  1  1  1  1  2  2  2  2  2  2  2  2  2  2  2  2  2  2  2  3
  3  3  3  3  3  3  3  3  3  3  3  3  3  3  3  3 39  3  4  4  4  4  4  4
  4  4  4  4  4  4  4  4  5  5  5  5  5  5  5  5  5  5  5  8  5  5  5  5
  5 34  6  6  6  6  6  0  6  6  6  6  6  6  6  6  8  7  7  7  7  7  7  7
  7  7  7  7  7  7  7  7 26  7  7  7  7  7  7  8  8  8  8  8  8  8  8  8
  8  8  8  8  8  8  8  8  8  8  8  9  9  9  9  9  9 36  9  9 36  9  9  9
  9  9  9  9 10 10 10 33 10 10 10 10 33 10 10 10 10 10 10 10 11 11 11 11
 11 11 11 11 11 11 11 11 11 11 11 11 12 12 12 12 12  1  1 12  1 12 12 12
 12 12 12 12 12 41 13 13 13 13 13 21 21 13 13 13 13 13 13 13 13 13 13 13
 14 14 14 14 14 14 14 14 14 14 14 14 14 14 18 14 14 14 15 29 40 15  5 15
  5 15 15 15 15 15 15 15 39 15 15 15 15  8  5 15 15 16 16 16 16 16  8 16
 16 16 16 16 16 16 16 16 16 29 16 34 16 17 17 17 17 17 17 17 17 17 17 17
 17 17 17 17 17 17 17 18 31 18 18 18 18 18 1

Les fichiers **label**  contiennent les ground truth. 

In [4]:
path = "/data/maouche/MultiOOD/HMDB-rgb-flow/saved_files/id_HMDB_label_a2d_npmix_best_val.npy"
data = np.load(path)
print(data.shape)
print(data)

(843,)
[ 0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  1  1  1  1  1  1  1  1
  1  1  1  1  1  1  1  1  2  2  2  2  2  2  2  2  2  2  2  2  2  2  2  3
  3  3  3  3  3  3  3  3  3  3  3  3  3  3  3  3  3  3  4  4  4  4  4  4
  4  4  4  4  4  4  4  4  5  5  5  5  5  5  5  5  5  5  5  5  5  5  5  5
  5  5  6  6  6  6  6  6  6  6  6  6  6  6  6  6  6  7  7  7  7  7  7  7
  7  7  7  7  7  7  7  7  7  7  7  7  7  7  7  8  8  8  8  8  8  8  8  8
  8  8  8  8  8  8  8  8  8  8  8  9  9  9  9  9  9  9  9  9  9  9  9  9
  9  9  9  9 10 10 10 10 10 10 10 10 10 10 10 10 10 10 10 10 11 11 11 11
 11 11 11 11 11 11 11 11 11 11 11 11 12 12 12 12 12 12 12 12 12 12 12 12
 12 12 12 12 12 13 13 13 13 13 13 13 13 13 13 13 13 13 13 13 13 13 13 13
 14 14 14 14 14 14 14 14 14 14 14 14 14 14 14 14 14 14 15 15 15 15 15 15
 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15 16 16 16 16 16 16 16
 16 16 16 16 16 16 16 16 16 16 16 16 17 17 17 17 17 17 17 17 17 17 17 17
 17 17 17 17 17 17 17 18 18 18 18 18 18 18 1

Les fichiers **output** contiennent les **logits**. 842 pour le nombre d'exemples et 43 pour le nombre de classes

In [5]:
path = "/data/maouche/MultiOOD/HMDB-rgb-flow/saved_files/id_HMDB_output_a2d_npmix_best_val.npy"
data = np.load(path)
print(data.shape)
print(data)

(843, 43)
[[  9.551006  -11.218089  -11.223899  ...  -7.963518   -6.9919205
   -8.55932  ]
 [  4.3062124 -11.8711    -11.465627  ...  -8.242423   -9.513602
   -6.135794 ]
 [  9.450798  -11.7504    -11.7918    ...  -7.7297983  -7.9175467
   -6.822984 ]
 ...
 [ -9.226287  -12.298967  -10.585246  ...  -7.449084  -11.904885
   10.397236 ]
 [ -4.497591  -11.434968   -8.496568  ...  -3.55011    -4.1009235
   -0.3867626]
 [-10.489369  -14.052661   -8.822249  ...  -6.165988  -11.213553
   12.434434 ]]


Les fichiers **conf** contiennent les scores de confiance, c-a-d les **max des softmax**

In [6]:
path = "/data/maouche/MultiOOD/HMDB-rgb-flow/saved_files/id_HMDB_conf_a2d_npmix_best_val.npy"
data = np.load(path)
print(data.shape)
print(data)

(843,)
[0.9999956  0.99937564 0.9999931  0.9999454  0.99964595 0.9751288
 0.9999999  1.         0.9999994  0.8223828  0.9996555  0.99999845
 0.99982196 0.8651386  0.9999982  1.         0.9912731  0.92672837
 0.5455324  0.99821717 1.         0.7314217  0.99301916 0.99979883
 0.9999956  0.5190761  0.9068743  0.99769753 0.9984653  0.999858
 0.9998797  0.809282   0.99996316 0.99999154 0.99999166 0.99998116
 0.99997187 0.99997604 0.99989736 0.9999999  0.999997   0.99458003
 0.98823404 0.9999924  0.99497825 0.9999641  0.9981255  1.
 1.         0.9999995  0.9999962  0.9999988  0.9999906  0.9999999
 0.99999857 0.99964595 0.99998534 1.         0.999977   1.
 0.9999981  0.9999989  0.9999989  0.99999976 0.64893293 0.99989235
 0.9999572  0.9998486  0.99973017 0.99935466 0.99769163 0.9388575
 0.70876926 0.9997515  0.99997723 0.9999584  0.9992725  0.9999701
 0.9998553  0.9998956  0.99999595 0.9787617  1.         0.99999726
 1.         0.9999999  0.99988914 0.99937254 0.9999683  0.9999962
 0.9999927 

In [ ]:
path = "/data/maouche/MultiOOD/EPIC-rgb-flow/saved_files/id_HMDB_ood_EPIC_conf_a2d_npmix_best_eval.npy"
data = np.load(path)
print(data.shape)
print(data)
print(data.mean())
print(data.std())

# Reconstitution des résultats

## HMDB ID vs UCF OOD eval

In [ ]:
config = "/data/maouche/MultiOOD/HMDB-rgb-flow/saved_files/id_HMDB_ood_{}_{}_a2d_npmix_best_eval.npy"
#"/data/maouche/MultiOOD/EPIC-rgb-flow/saved_files/id_HMDB_ood_EPIC_output_a2d_npmix_best_eval.npy"
ood_datasets = ['UCF', 'HAC']
cases = ["pred", "label", "output", "conf", "feature"]


Calculer les métriques

In [25]:
dataset = "UCF"
all_results_ucf = {}
for case in cases:
    path = config.format(dataset, case)
    all_results_ucf[case] = np.load(path)

#for case in cases:
print(all_results_ucf["label"])

[ 0  0  0 ... 94 94 94]


In [27]:
dataset = "UCF"
all_results_ucf = {}
for case in cases:
    path = config.format(dataset, case)
    all_results_ucf[case] = np.load(path)

fpr, auroc, aupr_in, aupr_out, accuracy = compute_all_metrics(all_results_ucf.get("conf",-1), 
                                            all_results_ucf.get("label", -1), all_results_ucf.get("pred", -1))

/home/maouche/.conda/envs/multiood/lib/python3.8/site-packages/sklearn/metrics/_ranking.py:1133: UndefinedMetricWarning: No positive samples in y_true, true positive value should be meaningless
  warnings.warn(
/home/maouche/.conda/envs/multiood/lib/python3.8/site-packages/sklearn/metrics/_ranking.py:980: UserWarning: No positive class found in y_true, recall is set to one for all thresholds.
  warnings.warn(


In [28]:
fpr, auroc, aupr_in, aupr_out, accuracy

(0.0, nan, 1.0, 0.5, 0.008882309400444115)

## réimplementation

In [53]:
ood_datasets = ["UCF", "EPIC", "HAC"]
pooling_types = ["max", "mean", "min"]

In [13]:
def compute_all_metrics(conf, label, pred):
    np.set_printoptions(precision=3)
    recall = 0.95
    auroc, aupr_in, aupr_out, fpr = auc_and_fpr_recall(conf, label, recall)

    accuracy = acc(pred, label)

    results = [fpr, auroc, aupr_in, aupr_out, accuracy]

    return results


# accuracy
def acc(pred, label):
    ind_pred = pred[label != -1]
    ind_label = label[label != -1]

    num_tp = np.sum(ind_pred == ind_label)
    acc = num_tp / len(ind_label)

    return acc


def auc_and_fpr_recall(conf, label, tpr_th):
    # following convention in ML we treat OOD as positive
    ood_indicator = label
    #ood_indicator = np.zeros(label)
    #ood_indicator[label == -1] = 1

    # in the postprocessor we assume ID samples will have larger
    # "conf" values than OOD samples
    # therefore here we need to negate the "conf" values
    fpr_list, tpr_list, thresholds = metrics.roc_curve(ood_indicator, -conf)
    fpr = fpr_list[np.argmax(tpr_list >= tpr_th)]

    precision_in, recall_in, thresholds_in \
        = metrics.precision_recall_curve(1 - ood_indicator, conf)

    precision_out, recall_out, thresholds_out \
        = metrics.precision_recall_curve(ood_indicator, -conf)

    auroc = metrics.auc(fpr_list, tpr_list)
    aupr_in = metrics.auc(recall_in, precision_in)
    aupr_out = metrics.auc(recall_out, precision_out)

    return auroc, aupr_in, aupr_out, fpr


In [ ]:
#auc_and_fpr_recall(all_results_ucf.get("conf",-1), all_results_ucf.get("label",-1), tpr_th=0.95)
# auroc, aupr_in, aupr_out, fpr

/home/maouche/.conda/envs/multiood/lib/python3.8/site-packages/sklearn/metrics/_ranking.py:1124: UndefinedMetricWarning: No negative samples in y_true, false positive value should be meaningless
  warnings.warn(
/home/maouche/.conda/envs/multiood/lib/python3.8/site-packages/sklearn/metrics/_ranking.py:980: UserWarning: No positive class found in y_true, recall is set to one for all thresholds.
  warnings.warn(


(nan, 0.5, 0.9999999999999999, nan)

Pour calculer la AUC et FPR, il faut les ground truth ID et OOD. Il faut donc combiner les résultats sur HMDB (j'ai pris le split val) avec les résultats sur le dataset OOD.
Aussi, dans leur fonction, ils mettent à 1 les valeurs égales à -1 dans les labels, sauf qu'en aucun cas on a un label de -1, donc avec leur implémentation on n'a jamais de label positif, d'où le warning

Là je fais de la **MSP** avec  npmix et a2d

In [12]:
def get_result(ood_dataset=None, case="conf", mode='val'):
    if ood_dataset == "EPIC":
        path = f"/data/maouche/MultiOOD/EPIC-rgb-flow/saved_files/id_HMDB_ood_EPIC_{case}_a2d_npmix_best_eval.npy"
    elif ood_dataset == None:
        path = f'/data/maouche/MultiOOD/HMDB-rgb-flow/saved_files/id_HMDB_{case}_a2d_npmix_best_{mode}.npy'
    else:
        path =  f"/data/maouche/MultiOOD/HMDB-rgb-flow/saved_files/id_HMDB_ood_{ood_dataset}_{case}_a2d_npmix_best_eval.npy"    
    return np.load(path)

In [ ]:
#combiner resultats HMDB et UCF pour avoir positifs et negatifs
#concaténer les conf et label

conf_hmdb = get_result(None, "conf", "val")
label_hmdb = get_result(None, "label")
conf_ucf = get_result("UCF", "conf")
label_ucf = get_result("UCF", "label")

conf_combine = np.hstack((conf_hmdb, conf_ucf))
label_combine = np.hstack((np.zeros_like(conf_hmdb), np.ones_like(label_ucf)))

In [57]:
auc_and_fpr_recall(conf_combine, label_combine, tpr_th=0.95)
# auroc, aupr_in, aupr_out, fpr

(0.889063327283599, 0.8536506758208082, 0.9162875427029971, 0.3973902728351127)

In [10]:
def compute_metrics(ood_dataset):
    conf_id = get_result(None, "conf", "val") #id est hmdb
    conf_ood = get_result(ood_dataset, "conf")
    pred_id = get_result(None, "pred", "val")
    pred_ood = get_result(ood_dataset, "pred")
    
    conf_combine = np.hstack((conf_id, conf_ood))
    label_combine = np.hstack((np.zeros_like(conf_id), np.ones_like(conf_ood)))
    pred_combine = np.hstack((pred_id, pred_ood))
    
    return compute_all_metrics(conf_combine, label_combine, pred_combine) 
    # fpr, auroc, aupr_in, aupr_out, accuracy


In [ ]:
results = {}

for dataset in ood_datasets:
    fpr, auroc, aupr_in, aupr_out, accuracy = compute_metrics(dataset)
    results.setdefault(dataset, {})
    results[dataset]["auroc"] = auroc
    results[dataset]["fpr"] = fpr

In [67]:
results

{'UCF': {'auroc': 0.889063327283599, 'fpr': 0.3973902728351127},
 'HAC': {'auroc': 0.9250053417622658, 'fpr': 0.27164887307236063},
 'EPIC': {'auroc': 0.9717896096514268, 'fpr': 0.12930011862396204}}

# Max/Mean/Min SP

Pour Mean et Min:
- récupérer les fichiers 'output' (logits) et appliquer un softmax
- Prendre le mean 

In [ ]:
output_hmdb = get_result(ood_dataset=None, case="output", mode='val')
#output_ood = get_result(ood_dataset="UCF", case="output")

sm_hmdb = torch.softmax(torch.tensor(output_hmdb), dim=1)
means_sm_hmdb = torch.mean(sm_hmdb, dim=1)
min_sm_hmdb = torch.min(sm_hmdb, dim=1)

In [39]:
def compute_metrics_sp(ood_dataset, pooling="max"): 
    #mode = {max, mean, min} max est équivalent à loader conf mais pour min et mean il faut loader output et faire softmax et prendre le min ou mean à la main
    assert pooling in {"max", "mean", "min"}
    
    if pooling == "max":
        conf_id = get_result(None, "conf", "val") #id est hmdb
        conf_ood = get_result(ood_dataset, "conf")

    else:
        output_id = get_result(ood_dataset=None, case="output", mode='val')
        output_ood = get_result(ood_dataset, case="output")
        
        sm_id = torch.softmax(torch.tensor(output_id), dim=1)
        sm_ood = torch.softmax(torch.tensor(output_ood), dim=1)
        if pooling == "mean":
            conf_id = torch.mean(sm_id, dim=1) 
            conf_ood = torch.mean(sm_ood, dim=1) 
            
        elif pooling == "min":
            conf_id = torch.min(sm_id, dim=1)[0]
            conf_ood = torch.min(sm_ood, dim=1)[0] 
            # print(conf_id)
            # print(conf_ood)
    
    pred_id = get_result(None, "pred", "val") # quand je fais le min, est ce que je prends comme pred le max ? jpense oui
    pred_ood = get_result(ood_dataset, "pred")
    
    conf_combine = np.hstack((conf_id, conf_ood))
    label_combine = np.hstack((np.zeros_like(conf_id), np.ones_like(conf_ood)))
    pred_combine = np.hstack((pred_id, pred_ood))
    
    
    return compute_all_metrics(conf_combine, label_combine, pred_combine) 
    # fpr, auroc, aupr_in, aupr_out, accuracy

In [54]:
results = {}

for pooling in pooling_types:
    results.setdefault(pooling, {})
    
    for dataset in ood_datasets:
        results.get(pooling,-1).setdefault(dataset, {})
        fpr, auroc, aupr_in, aupr_out, accuracy = compute_metrics_sp(dataset, pooling)
        #results.get(pooling,-1).get(dataset,-1).setdefault(dataset, {})
        results.get(pooling,-1).get(dataset,-1)["auroc"] = auroc
        results.get(pooling,-1).get(dataset,-1)["fpr"] = fpr

In [55]:
pprint.pp(results)

{'max': {'UCF': {'auroc': 0.889063327283599, 'fpr': 0.3973902728351127},
         'EPIC': {'auroc': 0.9717896096514268, 'fpr': 0.12930011862396204},
         'HAC': {'auroc': 0.9250053417622658, 'fpr': 0.27164887307236063}},
 'mean': {'UCF': {'auroc': 0.5238147920831896, 'fpr': 0.9596678529062871},
          'EPIC': {'auroc': 0.5172089980817062, 'fpr': 0.9596678529062871},
          'HAC': {'auroc': 0.5157263586232884, 'fpr': 0.9596678529062871}},
 'min': {'UCF': {'auroc': 0.07418870780661571, 'fpr': 1.0},
         'EPIC': {'auroc': 0.008952200461101636, 'fpr': 1.0},
         'HAC': {'auroc': 0.04851653385063345, 'fpr': 1.0}}}


# Phase 2

In [1]:
import pandas as pd

In [25]:
results_file = "results_postprocessings.csv"
df_results = pd.read_csv(results_file)

In [26]:
ordre_method = ['msp', 'ebo', 'maxlogit', 'Mahalanobis', 'react', 'ash', 'gen', 'knn', 'vim']
ordre_dataset = ['UCF', 'EPIC', 'HAC']
ordre_layer_proc = ['none', 'ash', 'react']


In [ ]:
df_results['method'] = pd.Categorical(df_results['method'], categories=ordre_method, ordered=True)
#df_results['dataset'] = pd.Categorical(df_results['dataset'], categories=ordre_dataset, ordered=True)
df_results['layer_proc'] = pd.Categorical(df_results['layer_proc'], categories=ordre_layer_proc, ordered=True)

# Trier d'abord par 'method', puis par 'dataset'
df_results = df_results.sort_values(['method', 'layer_proc']) #['method', 'dataset', 'layer_proc']
df_results.fpr95 = df_results.fpr95.apply(lambda x:x*100)
df_results.auroc = df_results.auroc.apply(lambda x:x*100)
df_results.id_acc = df_results.id_acc.apply(lambda x:x*100)

df_results.fpr95 = df_results.fpr95.apply(lambda x:np.round(x,3))
df_results.auroc = df_results.auroc.apply(lambda x:np.round(x,3))
df_results.id_acc = df_results.id_acc.apply(lambda x:np.round(x,3))


In [3]:
#df_results.to_csv("full_results.csv", index=False)
df_results = pd.read_csv("full_results.csv")

In [31]:
df_results

,method,dataset,layer_proc,fpr95,auroc,id_acc,exec_time
0,msp,UCF,none,40.593,88.000,86.773,NaN
1,msp,HAC,none,28.620,91.568,86.773,NaN
2,msp,EPIC,none,13.455,96.422,86.773,NaN
3,msp,UCF,ash,41.505,86.432,86.203,0:00:00.027182
4,msp,HAC,ash,28.734,90.328,86.203,0:00:00.042857
...,...,...,...,...,...,...,...
64,vim,HAC,ash,14.481,96.614,86.203,0:00:48.589382
65,vim,EPIC,ash,10.376,96.863,86.203,0:00:49.204083
66,vim,UCF,react,29.190,91.534,86.545,0:00:45.662581
67,vim,HAC,react,9.122,98.180,86.545,0:00:47.869842


In [8]:
vanilla_methods = df_results[df_results.layer_proc=="none"]
ash_react = df_results[df_results.method.isin(["ash", "react"])]

reimpl_papier = pd.concat([vanilla_methods, ash_react])
reimpl_papier['method'] = pd.Categorical(reimpl_papier['method'], categories=ordre_method, ordered=True)
reimpl_papier = reimpl_papier.sort_values(['method', 'layer_proc']) #['method', 'dataset', 'layer_proc']

reimpl_papier

,method,dataset,layer_proc,fpr95,auroc,id_acc,exec_time
6,msp,UCF,none,40.592930,88.000020,86.773090,NaN
7,msp,HAC,none,28.620296,91.567944,86.773090,NaN
8,msp,EPIC,none,13.454960,96.422314,86.773090,NaN
15,ebo,UCF,none,37.058153,89.712000,86.773090,NaN
16,ebo,HAC,none,23.033067,94.447204,86.773090,NaN
17,ebo,EPIC,none,7.183580,97.597912,86.773090,NaN
3,maxlogit,UCF,none,37.058153,89.728627,86.773090,NaN
4,maxlogit,HAC,none,23.147092,94.250283,86.773090,NaN
5,maxlogit,EPIC,none,7.069555,97.725678,86.773090,NaN
0,Mahalanobis,UCF,none,27.708096,91.210869,86.773090,NaN


In [ ]:
#reimpl_papier.to_csv("reimpl_papier.csv", index=False)
reimpl_papier = pd.read_csv("reimpl_papier.csv")

In [43]:
metrics_none = df_results[df_results.layer_proc=="none"][["method", "dataset", "fpr95", "auroc", "id_acc"]].reset_index()
metrics_ash = df_results[df_results.layer_proc=="ash"][["method", "dataset", "fpr95", "auroc", "id_acc"]].reset_index()
metrics_react = df_results[df_results.layer_proc=="react"][["method", "dataset", "fpr95", "auroc", "id_acc"]].reset_index()
metrics_diff = metrics_none.copy()

In [44]:
metrics_diff["ash_diff_fpr"] = metrics_ash.fpr95-metrics_diff.fpr95
metrics_diff["ash_diff_auroc"] = metrics_ash.auroc-metrics_diff.auroc
metrics_diff["ash_diff_id_acc"] = metrics_ash.id_acc-metrics_diff.id_acc
metrics_diff["react_diff_fpr"] = metrics_ash.fpr95-metrics_diff.fpr95
metrics_diff["react_diff_auroc"] = metrics_ash.auroc-metrics_diff.auroc
metrics_diff["react_diff_id_acc"] = metrics_ash.fpr95-metrics_diff.fpr95

In [47]:
metrics_diff

,index,method,dataset,fpr95,auroc,id_acc,ash_diff_fpr,ash_diff_auroc,ash_diff_id_acc,react_diff_fpr,react_diff_auroc,react_diff_id_acc
0,0,msp,UCF,40.593,88.000,86.773,0.912,-1.568,-0.57,0.912,-1.568,0.912
1,1,msp,HAC,28.620,91.568,86.773,0.114,-1.240,-0.57,0.114,-1.240,0.114
2,2,msp,EPIC,13.455,96.422,86.773,3.535,-2.105,-0.57,3.535,-2.105,3.535
3,9,ebo,UCF,37.058,89.712,86.773,1.711,-0.160,-0.57,1.711,-0.160,1.711
4,10,ebo,HAC,23.033,94.447,86.773,0.456,0.035,-0.57,0.456,0.035,0.456
5,11,ebo,EPIC,7.184,97.598,86.773,-0.228,0.653,-0.57,-0.228,0.653,-0.228
6,18,maxlogit,UCF,37.058,89.729,86.773,1.711,-0.231,-0.57,1.711,-0.231,1.711
7,19,maxlogit,HAC,23.147,94.250,86.773,0.456,0.098,-0.57,0.456,0.098,0.456
8,20,maxlogit,EPIC,7.070,97.726,86.773,0.228,0.374,-0.57,0.228,0.374,0.228
9,27,Mahalanobis,UCF,27.708,91.211,86.773,20.639,-10.357,-0.57,20.639,-10.357,20.639
